src/main/java/carpet/script/value/EntityValue.java

```diff
@@ -1668,12 +1690,56 @@ else if (v instanceof ListValue)
             }
         });
 
+        put("blue_skull",(e,v)->{
+            if (e instanceof WitherSkull w){
+                w.setDangerous(v.getBoolean());
+            }
+            
+        });
+        put("offering_flower",(e,v)->{
+            if (e instanceof IronGolem ig){
+                ig.offerFlower(v.getBoolean());
+            }
+            
+        });
+        put("item", (e, v) -> {
+                ItemStack item=getItemStackFromValue(v);
+                if(e instanceof ItemEntity itementity)            
+                    itementity.setItem(item);
+                if(e instanceof ItemFrame itemframe)
+                    itemframe.setItem(item);  
+        });
         // "dimension"      []
-        // "item"           []
         // "count",         []
         // "effect_"name    []
     }};
 
+    static public ItemStack getItemStackFromValue(Value value){
+        List<Value> items;
+        if(value instanceof ListValue blv && blv.length()==3){
+            items = blv.getItems();
+        }else if(value.isNull()){
+            return ItemStack.EMPTY;
+        }else{
+            items= List.of(StringValue.of(value.getString()),Value.ONE,Value.NULL);
+        }
+        String id=items.get(0).getString();
+        int count=((NumericValue)items.get(1)).getInt();
+        CompoundTag nbt=null;
+        if (!items.get(2).isNull()){
+            if(items.get(2) instanceof NBTSerializableValue nbtv){
+                nbt=nbtv.getCompoundTag();
+            }
+            else{
+                nbt=new NBTSerializableValue(items.get(2).getString()).getCompoundTag();
+            }
+        }
+        try{
+            return NBTSerializableValue.parseItem(id,nbt).createItemStack(count,false);
+        } catch (CommandSyntaxException e1) {
+            throw new IllegalStateException("Unexpected exception while creating item stack", e1);
```

Comment:
yeah, would throw INVALID_ITEM instead of illegal state. illegal state should not be thrown from invalid text input, allowing script writer to validate user input for example